tests/calc/files/displacements/test_lines.py

```diff
@@ -0,0 +1,454 @@
+        assert 'A' in rep and 'Fe' in rep and '0.0' in rep
+        assert 'B' in rep and 'Ni' in rep and '0.2' in rep
+        assert '=' in rep
+
+
+class TestConstraintLine:
+    @pytest.mark.parametrize(
+        'line, exp_type, exp_targets, exp_link_target, exp_op',
+        [
+            pytest.param(
+                'geo A_surf, A_def = linked',
+                TypeToken('geo'),
+                [TargetToken('A_surf'), TargetToken('A_def')],
+                TargetToken('A_surf'),
+                LinearOperationToken.from_array(np.eye(1)),
+                id='geo-linked-shorthand',
+            ),
+            pytest.param(
+                'geo A = B',
+                TypeToken('geo'),
+                [TargetToken('A')],
+                TargetToken('B'),
+                LinearOperationToken.from_array(np.eye(1)),
+                id='geo-direct-link',
+            ),
+            pytest.param(
+                'geo Fe 1 = Fe 2',
+                TypeToken('geo'),
+                [TargetToken('Fe 1')],
+                TargetToken('Fe 2'),
+                LinearOperationToken.from_array(np.eye(1)),
+                id='geo-single-target-link',
+            ),
+            pytest.param(
+                'geo A = [1 0 0] B',
+                TypeToken('geo'),
+                [TargetToken('A')],
+                TargetToken('B'),
+                LinearOperationToken('[1 0 0]'),
+                id='geo-linear-operation',
+            ),
+            pytest.param(
+                'geo Fe_surf, O_surf = [[1 0 0] [0 0 1] [0 1 0]] Fe_surf',
+                TypeToken('geo'),
+                [TargetToken('Fe_surf'), TargetToken('O_surf')],
+                TargetToken('Fe_surf'),
+                LinearOperationToken('[[1 0 0] [0 0 1] [0 1 0]]'),
+                id='geo-linear-operation',
+            ),
+            pytest.param(
+                'occ A, B = linked',
+                TypeToken('occ'),
+                [TargetToken('A'), TargetToken('B')],
+                TargetToken('A'),
+                LinearOperationToken.from_array(np.eye(1)),
+                id='occ-linked-shorthand',
+            ),
+            pytest.param(
+                'vib X, Y = 0.5 Z',
+                TypeToken('vib'),
+                [TargetToken('X'), TargetToken('Y')],
+                TargetToken('Z'),
+                LinearOperationToken.from_array(np.array(0.5)),
+                id='vib-float-link',
+            ),
+        ],
+    )
+    def test_valid_constraint_lines(
+        self, line, exp_type, exp_targets, exp_link_target, exp_op
+    ):
+        con = ConstraintLine(line)
+        # Type
+        assert con.type == exp_type
+        # Targets
+        if 'linked' in line:
+            for target, exp_target in zip(con.targets[:-1], exp_targets):
+                assert target == exp_target
+        else:
+            for target, exp_target in zip(con.targets, exp_targets):
+                assert target == exp_target
```

Comment:
```suggestion
        act_targets = con.targets
        if 'linked' in line:
            act_targets = act_targets[:-1]
        assert len(act_targets) == len(exp_targets)
        for target, exp_target in zip(act_targets, exp_targets):
            assert target == exp_target
```

Also, add a comment as to why `[:-1]`.